include/zenoh_commons.h

```diff
@@ -259,6 +259,40 @@ typedef struct z_owned_closure_hello_t {
   void (*call)(struct z_owned_hello_t*, void*);
   void (*drop)(void*);
 } z_owned_closure_hello_t;
+/**
+ * Owned variant of a Query received by a Queryable.
+ *
+ * You may construct it by `z_query_clone`-ing a loaned query.
+ * When the last `z_owned_query_t` corresponding to a query is destroyed, or the callback that produced the query cloned to build them returns,
+ * the query will receive its termination signal.
+ *
+ * Holding onto an `z_owned_query_t` for too long (10s by default, can be set in `z_get`'s options) will trigger a timeout error
+ * to be sent to the querier by the infrastructure, and new responses to the outdated query will be silently dropped.
+ */
+typedef struct z_owned_query_t {
+  void *_0;
+} z_owned_query_t;
```

Comment:
As per-naming convention, `z_x` means that is also available in `zenoh-pico`. However, this feature will initially only available in `zenoh-c`... should it (and all related functions) be called `zc_`?

Reply:
The plan was originally to introduce these symbols in pico during the release... They should arrive soon, but we could rename these for now until pico has them too.

Reply:
The rename would introduce a small break in zenoh-cpp, which would be a thorn in the release's side, so let's keep the `z_` prefix, with the goal of having pico finish up implementation by mid-January.

Reply:
Let's keep the name then to avoid excessive back-and-forth.